### PR TITLE
New: added environment variable support for all options in config.xml

### PR DESCRIPTION
**Pull request by sillock1**
#### Description
Adds environment variable support for all options in config.xml. Using the IOptions pattern this will allow users to override config with the following environment variables

The environment variables will take precedence over config.xml for populated values. This change is not breaking and will still default to config.xml for anything not set as an environment variable.

<details>
<summary>New environment variables</summary>

SONARR__APP__INSTANCENAME
SONARR__APP__THEME
SONARR__APP__LAUNCHBROWSER

SONARR__AUTH__APIKEY
SONARR__AUTH__ENABLED
SONARR__AUTH__METHOD
SONARR__AUTH__REQUIRED

SONARR__SERVER__URLBASE
SONARR__SERVER__BINDADDRESS
SONARR__SERVER__PORT
SONARR__SERVER__ENABLESSL
SONARR__SERVER__SSLPORT
SONARR__SERVER__SSLCERTPATH
SONARR__SERVER__SSLCERTPASSWORD

SONARR__LOG__LEVEL
SONARR__LOG__FILTERSENTRYEVENTS
SONARR__LOG__ROTATE
SONARR__LOG__SQL
SONARR__LOG__CONSOLELEVEL
SONARR__LOG__ANALYTICSENABLED
SONARR__LOG__SYSLOGSERVER
SONARR__LOG__SYSLOGPORT
SONARR__LOG__SYSLOGLEVEL

SONARR__UPDATE__MECHANISM
SONARR__UPDATE__AUTOMATICALLY
SONARR__UPDATE__SCRIPTPATH
SONARR__UPDATE__BRANCH
</details>

#### Database Migration
NO

#### Issues Fixed or Closed by this PR
[Closes](https://github.com/Sonarr/Sonarr/issues/6744)

### CODE REVIEW

**Comment by onedr0p** [comment]
Ref: https://github.com/Sonarr/Sonarr/issues/6744

File: src/NzbDrone.Core/Configuration/ConfigFileProvider.cs [hunk 1]
```diff
@@ -248,17 +282,29 @@ public string InstanceName
...
-        public bool UpdateAutomatically => GetValueBoolean("UpdateAutomatically", false, false);
+        public bool UpdateAutomatically => _updateOptions.Automatically ?? GetValueBoolean("UpdateAutomatically", false, false);
 
-        public UpdateMechanism UpdateMechanism => GetValueEnum("UpdateMechanism", UpdateMechanism.BuiltIn, false);
+        public UpdateMechanism UpdateMechanism
+        {
+            get
+            {
+                var isParsed = Enum.TryParse<UpdateMechanism>(_updateOptions.Mechanism, out var enumValue);
+                if (isParsed)
```
[the remark this answers is not in the document]
Comment (sillock1) [reply src/NzbDrone.Core/Configuration/ConfigFileProvider.cs:292]: Nope, just missed I could be more concise. I could go a step further and use the ternary conditional operator if you like?

`public UpdateMechanism UpdateMechanism =>
            Enum.TryParse<UpdateMechanism>(_updateOptions.Mechanism, out var enumValue)
                ? enumValue
                : GetValueEnum("UpdateMechanism", UpdateMechanism.BuiltIn, false)`

File: src/NzbDrone.Common/Options/HttpOptions.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+namespace NzbDrone.Common.Options;
+
+public class HttpOptions
+{
+    public string UrlBase { get; set; }
+    public string BindAddress { get; set; }
+    public int? Port { get; set; }
+    public bool? EnableSsl { get; set; }
```
Comment (markus101) [inline src/NzbDrone.Common/Options/HttpOptions.cs:8]: I think these should either be moved to `HttpsOptions` or `HttpOptions` needs to be renamed to something that is more generic and encompasses HTTPS things.

Comment (bo0tzz) [reply src/NzbDrone.Common/Options/HttpOptions.cs:8]: Perhaps `ServerOptions`?

Comment (markus101) [reply src/NzbDrone.Common/Options/HttpOptions.cs:8]: I think that works, was the first one that came to mind for me too

Comment (sillock1) [reply src/NzbDrone.Common/Options/HttpOptions.cs]: I've opted to go with ServerOptions 👍

**Comment by sillock1** [comment]
Are any of these failing tests related to the pr changes or are they flaky tests?

**Pull request merged**